### PR TITLE
[feature]support configurable type and annotations to MariaDB service

### LINKED ISSUE #46: `ServiceTemplate` to provide `Service` type and annotations
Be able to provide a `Service` template in the `MariaDB` CRD to customize how it is exposed in the cluster. For example, if we wanted to expose it using [Metallb](https://metallb.universe.tf/usage/example/):
```yaml
apiVersion: mariadb.mmontes.io/v1alpha1
kind: MariaDB
metadata:
  name: mariadb
spec:
    ....
    service:
       type: LoadBalancer
       annotations:
            metallb.universe.tf/address-pool: sandbox
```

### PR DESCRIPTION
**Pull request by ksankeerth**
Hi @mmontes11 

Please check this PR and let me know anything to be changed. I added modified changes of  generated files as well. I tested locally and it seems to be working.  Service is considered as optional. If we don't specify, I pass an empty struct to client. Since K8s default is ClusterIP, it'll create a service with ClusterIP.  

Fixes #46 

Thanks,
Sankeerthan

### CODE REVIEW

**Review by mmontes11 — CHANGES_REQUESTED** [review]
Hey @ksankeerth ! thanks a lot for your contribution ❤️ 🙇🏻 

The changes LGTM 🚀 , left a suggestion, could you take a look?

Could you also please add some test cases?:
- Specify a `Service` [here](https://github.com/mariadb-operator/mariadb-operator/blob/main/controllers/suite_test_data.go#L134) to make sure it reconciles properly
- Add a case [here](https://github.com/mariadb-operator/mariadb-operator/blob/main/api/v1alpha1/mariadb_webhook_test.go#L313) to test that the `Service` is mutable

File: api/v1alpha1/mariadb_types.go [hunk 1]
```diff
@@ -42,6 +42,11 @@ type Metrics struct {
 	ServiceMonitor ServiceMonitor `json:"serviceMonitor"`
 }
 
+type MariaDBService struct {
```
[the remark this answers is not in the document]
Comment (ksankeerth) [reply api/v1alpha1/mariadb_types.go]: +1. done the changes.

**Comment by ksankeerth** [comment]
> Hey @ksankeerth ! thanks a lot for your contribution ❤️ 🙇🏻
> 
> The changes LGTM 🚀 , left a suggestion, could you please take a look?
> 
> Could you also please add some test cases?:
> 
> * Specify a `Service` [here](https://github.com/mariadb-operator/mariadb-operator/blob/main/controllers/suite_test_data.go#L134) to make sure it reconciles properly
> * Add a case [here](https://github.com/mariadb-operator/mariadb-operator/blob/main/api/v1alpha1/mariadb_webhook_test.go#L313) to test that the `Service` is mutable

@mmontes11  Thanks for  the review. I have done the changes requested. Please check when you have time.

**Review by mmontes11 — APPROVED** [review]
LGTM 🚀  Thanks for contributing to `mariadb-operator` 🦭 ❤️

**Pull request merged**